### PR TITLE
etcdserver: candidate should wait for applying all configuration changes

### PR DESCRIPTION
**Pull request by xiang90**
Fix https://github.com/coreos/etcd/issues/7280

/cc @siddontang

### CODE REVIEW

File: etcdserver/server.go [hunk 1]
```diff
@@ -608,6 +608,7 @@ type etcdProgress struct {
 type raftReadyHandler struct {
 	updateLeadership     func()
 	updateCommittedIndex func(uint64)
+	waitForApplyToFinish func()
```
Comment (heyitsanthony) [inline etcdserver/server.go]: `drainApply`? `waitForApply`? seems wordy

**Comment by heyitsanthony** [comment]
lgtm; is there any simple way to test this?

File: etcdserver/server.go (hunk 1 above)
Comment (xiang90) [inline etcdserver/server.go]: OK. probably just waitForApply

**Comment by xiang90** [comment]
@heyitsanthony I will try to figure out a test case.

**Comment by siddontang** [comment]
LGTM

**Comment by xiang90** [comment]
@heyitsanthony A test case is added.

**Comment by heyitsanthony** [comment]
lgtm thanks!

**Pull request merged**